### ISSUE TITLE
Same dependency inside and outside of @defer throws an error 

### ISSUE THREAD

**Issue by JeanMeche** [angular/angular#52012]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

The following template throws NG0300: 
```
@defer(on immediate) { 
   @for (item of items; track item.id) {
     <item [id]="item.id" />
    } 
} 
@for (item of items; track item.id) {
 <item [id]="item.id" />
} 
```

Here we have a template where the same dependency is located inside and outside the `@defer`. 

Note: It might be fixed by #51964, but not sure.  

[Demo](https://stackblitz.com/edit/angular-at-qguzbb?file=src%2Fmain.ts)

### Please provide the environment you discovered this bug in (run `ng version`)

```true
v17.0.0-next.6
```

**Comment by JoostK** [comment]
Indeed, this should get resolved once #51964 gets in.

**Comment by crisbeto** [comment]
Looks like the linked PR is merged. Closing this issue.
